ludwig/utils/image_utils.py

```diff
@@ -107,14 +108,35 @@ def is_image_score(src_path, img_entry):
 
 
 @functools.lru_cache(maxsize=32)
-def read_image(img: Union[str, torch.Tensor], num_channels: Optional[int] = None) -> torch.Tensor:
-    """ Returns a tensor with CHW format.
+def read_image(img: Union[str, bytes, BytesIO, torch.Tensor], num_channels: Optional[int] = None) -> torch.Tensor:
+    """Returns a tensor with CHW format.
 
-    If num_channels is not provided, he image is read in unchanged format.
+    If num_channels is not provided, the image is read in unchanged format.
+    Returns None if the image could not be read.
     """
+    if isinstance(img, torch.Tensor):
+        return img
     if isinstance(img, str):
         return read_image_from_str(img, num_channels)
-    return img
+    if isinstance(img, bytes):
+        with BytesIO(img) as buffer:
+            buffer_view = buffer.getbuffer()
```

Comment:
what happens if the img is too big to fit in the buffer?

Reply:
I imagine that the entire program would terminate with OOM. Hopefully no one's reading in images that are THAT big.